### DISCUSSION TITLE
How not to create `.classpath, .project and .settings/*` files at each module's root dir?

### DISCUSSION THREAD

**Discussion by devgony** [mfussenegger/nvim-jdtls#724] [Q&A]
Currenlty, whenever i open nvim on java files with nvim-jdtls, it creates `.classpath, .project and .settings/*` files at each module's 
root dir.

How can i configure it to be created out of java project source?
i know vscode-jdtls creates those files on some temporary directory like `~/Library/Application Support/Code/User/workspaceStorage/`

**Comment by Skylli202** [comment]
+1

Have you been able to work around this, @devgony?

I assume I'm in a similar situation to you; these are not in the `.gitignore` files of my company repositories, and I cannot easily add them; I'd rather them be stored in `~/.cache/jdtls/[...]` like the rest of it.

**Reply by Desoroxxx** [reply]
As a workaround, you can use https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreexcludesFile, which allows you to have a global `.gitignore` for your entire personal machine.

**Reply by devgony** [reply]
Sorry to reply late.
As @Desoroxxx mentioned, I used to set global gitignore!

**Comment by Skylli202** [comment]
I've found https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/1900 which we could use but have not been successful yet in setting it up in the nvim-jdtls setup (lazyvim with lazyextra lang.java).
